python/lsst/analysis/tools/tasks/catalogMatch.py

```diff
@@ -125,10 +125,38 @@ class CatalogMatchConfig(pipeBase.PipelineTaskConfig, pipelineConnections=Catalo
         default=1.0,
     )
 
-    raColumn = pexConfig.Field[str](doc="RA column.", default="coord_ra")
-    decColumn = pexConfig.Field[str](doc="Dec column.", default="coord_dec")
+    targetRaColumn = pexConfig.Field[str](
+        doc="RA column name for the target (being matched) catalog.",
+        default="coord_ra",
+    )
+
+    targetDecColumn = pexConfig.Field[str](
+        doc="Dec column name for the target (being matched) catalog.",
+        default="coord_dec",
+    )
+
+    refRaColumn = pexConfig.Field[str](
+        doc="RA column name for the reference (being matched to) catalog.",
+        default="ra",
+    )
+
+    refDecColumn = pexConfig.Field[str](
+        doc="Dec column name for the reference (being matched to) catalog.",
+        default="dec",
+    )
+
     patchColumn = pexConfig.Field[str](doc="Patch column.", default="patch")
 
+    refCat = pexConfig.Field[bool](
+        doc="Is the catalog being matched to a reference catalog?",
```

Comment:
I'm confused as to what this actually means. The flag only seems to rename reference magnitude columns if enabled. So should it be disabled when the refCatalog (the one being matched to) doesn't have magnitudes at all (or only fluxes? or neither?), or the naming scheme doesn't match expectations? When would any of those conditions be the case?

Separately, when you say "reference catalog" do you specifically mean a reference catalog used for calibration distinctly from the `refCatalog` being matched?

Reply:
It wants to know if the catalog being matched to is stored as a reference catalog because then the fluxes need to be flattened out.